pkg/common/endpoint.go

```diff
@@ -1,12 +1,25 @@
 // Copyright (c) Microsoft Corporation.
-// Licensed under the MIT license.
+// Licensed under the MIT licensep.
 package common
 
 import (
+	"errors"
 	"fmt"
 	"net"
 )
 
+var ErrNoIPFoundForEndpoint = errors.New("no IP found for endpoint")
+
+func noIPFoundForEndpointError(endpointKey string) error {
+	return fmt.Errorf("%w %s", ErrNoIPFoundForEndpoint, endpointKey)
```

Comment:
instead of string fmt, let's wrap

```suggestion
	return errors.Wrap(ErrNoIPFoundForEndpoint, endpointKey)
```
applies to all of this pattern in this change. We wrap so that comparisons via `errors.Is` and `errors.As` in the calling code will work to tell us if a child error has one of your typed errors in its heritage. 
With fmt, `errors.Is(err, ErrNoIPFoundForEndpoint)` is always false, because fmt.Errorf is making a new, unique error every time.

Reply:
Thanks for your feedback~ :smiley: I have applied the `errors.Wrap` method and squashed this PR. Looking forward for any more suggestions 

P.s.
With the `%w` verb, the `fmt.Errorf` wraps the errors as well. But the helper functions are unnecessary.
![image](https://github.com/microsoft/retina/assets/22489650/2eaa5470-60bc-4947-9bb7-0f446513c970)

Reply:
Ah you're right about `%w`, I'm sorry! I give this feedback often (`fmt`->`errors`) and didn't process the fmt verbs here 🙈